genstudio-mlr-claims-app/src/genstudiopem/web-src/src/components/RightPanel.tsx

```diff
@@ -90,108 +141,144 @@ export default function RightPanel(): JSX.Element {
     setIsPolling(false);
   };
 
-  useEffect(() => {
-    if (guestConnection) {
-      pollForExperiences();
-    }
-  }, [guestConnection]);
+  const renderClaimsLibraryPicker = () => (
+    <Picker
+      label="Select claim library"
+      width="100%"
+      onSelectionChange={handleClaimsLibrarySelection}
+    >
+      {TEST_CLAIMS.map(library => (
+        <Item key={library.id}>{library.name}</Item>
+      ))}
+    </Picker>
+  );
+
+  const renderExperiencePicker = () => {
+    if (!experiences) return null;
+
+    return (
+      <Picker
+        label="Select experience"
+        align="start"
+        isDisabled={!selectedClaimLibrary || isSyncing}
+        onSelectionChange={handleExperienceSelection}
+      >
+        {experiences.map((experience, index) => (
+          <Item key={experience.id}>{`Experience ${index + 1}`}</Item>
+        ))}
+      </Picker>
+    );
+  };
+
+  const renderSyncExperiencesButton = () => (
+    <Button
+      variant="secondary"
+      width="160px"
+      isDisabled={isSyncing}
+      onPress={getExperience}
+      marginTop="size-300"
+    >
+      {isSyncing ? (
+        <Flex
+          direction="row"
+          alignItems="center"
+          justifyContent="space-around"
+          gap="size-50"
+        >
+          <ProgressCircle
```

Comment:
note: the reason i was using progressCircle here instead of spinner was to get it small enough. if you can get it working with the spinner component that would be more ideal

Reply:
maybe not hugely important since we want to remove this in next PR